### PR TITLE
Implementing zip data store and sdk fetch logic

### PR DESCRIPTION
**Pull request by andrerod**

### CODE REVIEW

File: lib/hashedpackage.js [hunk 1]
```diff
@@ -46,21 +46,23 @@ _.extend(HashedPackage.prototype, {
   /**
   * Adds a content to the package data store.
   *
-  * @param {string} store            The store where to add (i.e. LocalContent, NamedStreams, etc).
-  * @param {string} [name]           The name of the content definition. If null, a guid will be used instead.
-  * @param {string} originalFullPath The full path to the file to add.
+  * @param {string} store           The store where to add (i.e. LocalContent, NamedStreams, etc).
+  * @param {string} [name]          The name of the content definition. If null, a guid will be used instead.
+  * @param {string} origin          The full path to the file to add.
```
Comment (christav) [inline lib/hashedpackage.js]: Is this a string or an object, or are you tacking extra properties onto a string?

File: lib/sdkpackage.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  utils.validateArgs('SDKPackage', function (v) {
+    v.object(options, 'options');
+  });
+
+  this.cachePath = options.cachePath;
+}
+
+_.extend(SDKPackage.prototype, {
+  /**
+  * Retrives the list of files from the sdk
```
Comment (christav) [inline lib/sdkpackage.js]: spelling - "Retrieves"

File: lib/util/constants.js [hunk 3]
```diff
@@ -29,5 +31,11 @@ _.extend(exports, {
 
   IntegrityCheckHashAlgortihms: {
     Sha256: 'Sha256'
+  },
+
+  DefaultSDKLocation: {
+    // TODO: Move to azure storage at some point
+    manifest: 'https://raw.github.com/andrerod/cspack/dev3/lib/pkg/sdk.json',
```
Comment (christav) [inline lib/util/constants.js:38]: What about the versioning for the SDK? Is there ever a case where a package will work with SDK version n, but not with SDK version n + 1?

File: lib/util/utils.js [hunk 4]
```diff
@@ -98,32 +101,60 @@ _.extend(exports, {
               return callback(err1);
             }
 
-            execute(dirs.join(path.sep), root, callback);
+            execute(dirs, root, callback);
```
Comment (christav) [inline lib/util/utils.js]: execute function doesn't take a callback parameter?

File: lib/util/utils.js [hunk 5]
```diff
@@ -98,32 +101,60 @@ _.extend(exports, {
...
-  * @param {string} filename         The file name.
+  * @param {string} origin           The file name or the content.
+  * @param {string} origin.filePath  The origin file path.
+  * @param {string} origin.content   The origin content.
   * @param {string} hashingAlgorithm The hashing algorithm (e.g. sha256)
   * @param {function(err, checksum)} callback The file checksum.
   */
-  calculateFileHashsum: function (filename, hashingAlgorithm, callback) {
+  calculateFileHashsum: function (origin, hashingAlgorithm, callback) {
     var shasum = crypto.createHash(hashingAlgorithm);
```
Comment (christav) [inline lib/util/utils.js:151]: Since algorithm isn't always required to be a SHA family hash, this should probably be named "sum" or "hash".

**Comment by christav** [comment]
Minor Nitpicks, LGTM

File: lib/hashedpackage.js (hunk 1 above)
Comment (andrerod) [inline lib/hashedpackage.js]: It's an object.

File: lib/util/constants.js (hunk 3 above)
Comment (andrerod) [inline lib/util/constants.js:38]: Maybe. But i'm not covering that scenario for now.

File: lib/util/utils.js (hunk 4 above)
Comment (andrerod) [inline lib/util/utils.js]: Will fix.

File: lib/util/utils.js (hunk 5 above)
Comment (andrerod) [inline lib/util/utils.js:151]: Will fix.

**Pull request merged**